apps/server/src/modules/h5p-library-management/service/h5p-library-management.service.ts

```diff
@@ -134,60 +156,366 @@ export class H5PLibraryManagementService {
-		this.checkContentTypeExists(contentType);
 
-		const user = this.createDefaultIUser();
+		return tags;
+	}
+
+	private async installLibraryTagFromGitHub(library: string, tag: string): Promise<void> {
+		const tempFolder = tmpdir();
+		const libraryName = library.split('/')[1];
+		const filePath = join(tempFolder, `${libraryName}-${tag}.zip`);
+		const folderPath = join(tempFolder, `${libraryName}-${tag}`);
 
-		let installResults: ILibraryInstallResult[] = [];
+		await this.downloadGitHubTag(library, tag, filePath);
+		H5PLibraryHelper.unzipFile(filePath, tempFolder);
+		this.checkAndCorrectLibraryJsonVersion(folderPath, tag);
+		if (
+			(library === 'h5p/h5p-drag-question' && tag === '1.13.15') ||
+			(library === 'h5p/h5p-course-presentation' && tag === '1.25.33') ||
+			(library === 'h5p/h5p-interactive-video' && tag === '1.26.40') ||
+			(library === 'h5p/h5p-editor-audio-recorder' && tag === '1.0.12')
+		) {
+			this.buildLibraryIfRequired(folderPath, library);
+		}
+		if (library === 'h5p/h5p-memory-game' && tag === '1.3.36') {
+			this.checkAndCorrectLibraryJsonPaths(folderPath);
+		}
 
 		try {
-			installResults = await this.contentTypeRepo.installContentType(libraryToBeInstalled, user);
+			await this.libraryManager.installFromDirectory(folderPath);
 		} catch (error: unknown) {
-			this.logger.warning(new H5PLibraryManagementErrorLoggable(libraryToBeInstalled, error));
+			const loggableError =
+				error instanceof Error
+					? new H5PLibraryManagementErrorLoggable(library, error)
+					: new H5PLibraryManagementErrorLoggable(library, new Error('Unknown error during installation'));
+			this.logger.warning(loggableError);
 		}
+		H5PLibraryHelper.removeTemporaryFiles(filePath, folderPath);
+	}
 
-		const installedLibraries = await this.installLibraries(
-			librariesToInstall.slice(0, lastPositionLibrariesToInstallArray)
-		);
+	private buildLibraryIfRequired(folderPath: string, library: string): void {
+		const packageJsonPath = join(folderPath, 'package.json');
+		if (existsSync(packageJsonPath)) {
+			this.logger.info(new H5PLibraryManagementLoggable(`Running npm ci and npm run build in ${folderPath}`));
+			const npmInstall = spawnSync('npm', ['ci'], { cwd: folderPath, stdio: 'inherit' });
+			if (npmInstall.status !== 0) {
+				this.logger.warning(new H5PLibraryManagementErrorLoggable(library, new Error('npm ci failed')));
+			} else {
+				const npmBuild = spawnSync('npm', ['run', 'build'], { cwd: folderPath, stdio: 'inherit' });
+				if (npmBuild.status !== 0) {
+					this.logger.warning(new H5PLibraryManagementErrorLoggable(library, new Error('npm run build failed')));
+				}
+				const nodeModulesPath = join(folderPath, 'node_modules');
+				if (existsSync(nodeModulesPath)) {
+					spawnSync('rm', ['-rf', 'node_modules'], { cwd: folderPath, stdio: 'inherit' });
+					this.logger.info(new H5PLibraryManagementLoggable(`Removed node_modules from ${folderPath}`));
+				}
+			}
+		}
+	}
```

Comment:
Das sieht nicht so aus als ob es hier hingehört. Sollte verschoben werden in eine extra Datei. 
Wie gut kommt ein pod damit zurecht wenn du dort npm run build ausführst? Von Natur des Designes her sind sie dafür nicht ausgelegt.  
Das sollte unbedingt noch mal mit DevOps abgesprocen werden. 

Ich kann mir vorstellen das es im Fehlerfall zu unerwünschten Verhalten führt.